go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -1373,3 +1388,172 @@ func configureGlobalForwarding() error {
+func deleteMasqueradeResources(link netlink.Link, staleMasqueradeIPs *config.MasqueradeIPsConfig) error {
+	var subnets []*net.IPNet
+	var neighborIPs []net.IP
+	if config.IPv4Mode && staleMasqueradeIPs.V4HostMasqueradeIP != nil {
+		_, masqIPNet, err := net.ParseCIDR(fmt.Sprintf("%s/32", staleMasqueradeIPs.V4OVNMasqueradeIP.String()))
+		if err != nil {
+			return fmt.Errorf("failed to parse V4OVNMasqueradeIP %s: %v", staleMasqueradeIPs.V4OVNMasqueradeIP.String(), err)
+		}
+		subnets = append(subnets, masqIPNet)
+		neighborIPs = append(neighborIPs, staleMasqueradeIPs.V4OVNMasqueradeIP, staleMasqueradeIPs.V4DummyNextHopMasqueradeIP)
+		if err := nodeipt.DelRules(getStaleMasqueradeIptablesRules(staleMasqueradeIPs.V4OVNMasqueradeIP)); err != nil {
+			return fmt.Errorf("failed to delete forwarding iptables rules for stale masquerade subnet %s: ", err)
+		}
+		// Delete any stale masquerade IP from external bridge.
+		hostMasqIPNet, err := util.LinkAddrGetIPNet(link, staleMasqueradeIPs.V4HostMasqueradeIP)
+		if err != nil {
+			return fmt.Errorf("unable to get IPNet from link %s: %v", link, err)
+		}
+		if err := util.LinkAddrDel(link, hostMasqIPNet); err != nil {
+			return fmt.Errorf("failed to remove masquerade IP from bridge %s: %w", link, err)
+		}
+	}
+
+	if config.IPv6Mode && staleMasqueradeIPs.V6HostMasqueradeIP != nil {
+		_, masqIPNet, err := net.ParseCIDR(fmt.Sprintf("%s/128", staleMasqueradeIPs.V6OVNMasqueradeIP.String()))
+		if err != nil {
+			return fmt.Errorf("failed to parse V6OVNMasqueradeIP %s: %v", staleMasqueradeIPs.V6OVNMasqueradeIP.String(), err)
+		}
+		subnets = append(subnets, masqIPNet)
+		neighborIPs = append(neighborIPs, staleMasqueradeIPs.V6OVNMasqueradeIP, staleMasqueradeIPs.V6DummyNextHopMasqueradeIP)
+		if err := nodeipt.DelRules(getStaleMasqueradeIptablesRules(staleMasqueradeIPs.V6OVNMasqueradeIP)); err != nil {
+			return fmt.Errorf("failed to delete forwarding iptables rules for stale masquerade subnet %s: ", err)
+		}
+		// Delete any stale masquerade IP from external bridge.
+		hostMasqIPNet, err := util.LinkAddrGetIPNet(link, staleMasqueradeIPs.V6HostMasqueradeIP)
+		if err != nil {
+			return fmt.Errorf("unable to get IPNet from link %s: %v", link, err)
+		}
+		if err := util.LinkAddrDel(link, hostMasqIPNet); err != nil {
+			return fmt.Errorf("failed to remove masquerade IP from bridge %s: %w", link, err)
+		}
+	}
+
+	for _, ip := range neighborIPs {
+		if err := util.LinkNeighDel(link, ip); err != nil {
+			klog.Warningf("Failed to remove IP neighbour entry for ip %s, on iface %s: %v",
+				ip, link.Attrs().Name, err)
+		}
+	}
+
+	if len(subnets) != 0 {
+		if err := util.LinkRoutesDel(link, subnets); err != nil {
+			return fmt.Errorf("failed to list addresses for the link %s: %v", link.Attrs().Name, err)
+		}
+	}
+	return nil
+}
+
+// getOldMasquerdeSubnet determines the old masquerade subnet if it has changed at day 2.
+// It evaluates routes for ovn-k8s-mp0 interface to determine old subnet.
```

Comment:
So the old masq subnet is getting extracted from a route? Didn't we discuss looking it up in OVN? https://redhat-internal.slack.com/archives/C077NGF5HE1/p1720531355364399?thread_ts=1720529597.653169&cid=C077NGF5HE1

Reply:
I evaluated that option but DefaultNodeNetworkController do not have nbClient.

Reply:
resolving all conversations due to change in approach.